src/extension.ts

```diff
@@ -2,28 +2,33 @@
 // The module 'vscode' contains the VS Code extensibility API
 // Import the module and reference it with the alias vscode in your code below
 import * as vscode from 'vscode';
-
 import * as utilities from './utilities';
-import * as dockerRunner from './dockerRunner';
-import * as terminalExecutor from './terminalExecutor';
-import * as cloudshellRunner from './cloudShellRunner';
 import { AzureAccount } from './azure-account.api';
 import { Constants } from './constants';
+import { TerminalRunner } from './terminalRunner';
+import { CloudShellRunner } from './cloudShellRunner';
+import { TerminalExecutor } from './terminalExecutor';
+
 
 export function activate(context: vscode.ExtensionContext) {
     console.log('Congratulations, your extension "vscode-ansible" is now active!');
     var outputChannel = vscode.window.createOutputChannel("VSCode extension for Ansible");
+    utilities.generateCredentialsFile();
+
+    var terminalRunner = new TerminalRunner(outputChannel);
+    var cloudShellRunner = new CloudShellRunner(outputChannel);
 
-    let playbookinterminal = vscode.commands.registerCommand('vscode-ansible.ansible-playbook-in-terminal', () => {
-        utilities.runPlaybookInTerminal();
-    })
+    context.subscriptions.push(vscode.commands.registerCommand('vscode-ansible.playbook-in-terminal', () => {
+        terminalRunner.runPlaybook();
+    }));
 
-    let runcloudshell = vscode.commands.registerCommand('vscode-ansible.ansible-cloudshell', () => {
-        cloudshellRunner.runPlaybook(outputChannel);
-    });
+    context.subscriptions.push(vscode.commands.registerCommand('vscode-ansible.cloudshell', () => {
+        cloudShellRunner.runPlaybook();
+    }));
 
-    context.subscriptions.push(playbookinterminal);
-    context.subscriptions.push(runcloudshell);
+    context.subscriptions.push(vscode.window.onDidCloseTerminal((closedTerminal: vscode.Terminal) => {
```

Comment:
why is this needed here?
i thought it was enough to have it in TerminalExecutor, as i have tested it was called every time when terminal was closed.
is there any additional reason for this code?

Reply:
terminalExecutor is changed, it is just a helper class. here call vscode.window.onDidCloseTerminal